app/workers/facilities/state_cemetery_reload_job.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Facilities
+  class StateCemeteryReloadJob
+    include Sidekiq::Worker
+
+    def perform
+      ActiveRecord::Base.transaction do
+        purge_cemeteries
+        fetch_cemeteries.xpath('//cems//cem').map do |cemetery|
+          next if cemetery['lat'].blank? || cemetery['long'].blank?
+          Facilities::NCAFacility.create build_attributes(cemetery)
+        end
+      end
+    end
+
+    private
+
+    def purge_cemeteries
+      Facilities::NCAFacility.where(classification: 'State Cemetery').delete_all
+    end
+
+    def fetch_cemeteries
+      # our copy will become the source of truth but it is based off of:
+      # Nokogiri::HTML(open("https://www.cem.va.gov/cems/cems.xml"))
+      File.open(Rails.root.join('lib', 'facilities', 'cemetery_data', 'cems.xml')) do |file|
+        Nokogiri::XML(file)
+      end
+    end
+
+    def build_attributes(cemetery)
+      { unique_id: "s#{cemetery['fac_id']}",
+        name: cemetery['cem_name'],
+        classification: 'State Cemetery',
+        website: cemetery['cem_url'],
+        lat: cemetery['lat'],
+        long: cemetery['long'],
+        address: {
+          'physical' => parse_address(cemetery['address_line1'], cemetery['address_line2'], cemetery['address_line3']),
+          'mailing' => parse_address(cemetery['mailing_line1'], cemetery['mailing_line2'], cemetery['mailing_line3'])
+        },
+        phone: {
+          'main' => cemetery['phone'],
+          'fax' => cemetery['fax']
+        },
+        hours: hours,
+        services: {},
+        feedback: {},
+        access: {} }
+    end
+
+    def hours
+      { 'Monday' => 'Sunrise - Sunset',
+        'Tuesday' => 'Sunrise - Sunset',
+        'Wednesday' => 'Sunrise - Sunset',
+        'Thursday' => 'Sunrise - Sunset',
+        'Friday' => 'Sunrise - Sunset',
+        'Saturday' => 'Sunrise - Sunset',
+        'Sunday' => 'Sunrise - Sunset' }
+    end
+
+    def parse_address(part1, part2, part3)
+      return nil if part1.blank?
+      if part2.blank?
+        return part1 =~ /, / ? city_state_zip(part1) : { 'address_1' => part1 }
+      end
+      if part3.blank?
```

Comment:
Could this just be an `elif`? Seems like it would be more consistent than having the explicit return in the `part2.blank?`.

Reply:
yeah, thanks for pointing out.. I started when something way more complex and gave up too early when cleaning it up!